### ISSUE TITLE
Add collision resolution

### ISSUE THREAD

**Issue by AlexBezuska** [SplatJS/splat-ecs#19]
Axis Aligned Bounding Box 
(direct replacement for the one we use commonly)
https://www.npmjs.com/package/aabb-2d
https://www.npmjs.com/package/aabb-3d
https://www.npmjs.com/package/aabb

**Comment by AlexBezuska** [comment]
Lots of other interesting collision modules:
https://www.npmjs.com/search?q=collision

**Comment by CaldwellYSR** [comment]
How would we differentiate collisions and triggers? Would we create a new triggers component to handle non-physical collisions or however you say that?

**Comment by CaldwellYSR** [comment]
https://www.npmjs.com/package/sat

This package looks pretty cool. Support circles and polygons

**Comment by CaldwellYSR** [comment]
Would there be any interest in having a collision detection system that supports shapes besides rectangles? I imagine it would be a bear to overhaul the collision system but I think having other polygons would be pretty nice.

**Comment by ericlathrop** [comment]
Yes. It's a lot of work, though. You'll want to Google separating axis theorem. There might be an npm module for it.

**Comment by CaldwellYSR** [comment]
Yeah the package I posted a while back was an SAT package. Looked simple enough.

**Comment by CaldwellYSR** [comment]
https://github.com/CaldwellYSR/splat-ecs/blob/collisions/lib/systems/box-collider.js#L31

This is using SAT to replace the box-collider but it's got a couple problems. It seems to report collisions correctly for the most part but I was also finding situations where idA and idB were the same like it was reporting collisions between the same entity. On top of that, I kept getting (cannot find "collisions" of undefined) which I think was happening when an entity was being destroyed in the middle of trying to update the collisions array, I could be wrong though.

I think I need a faster way to test collisions between all the elements besides that double for loop.
